.github/workflows/build-gem.yml

```diff
@@ -0,0 +1,138 @@
+name: Build gem
+
+on:
+  workflow_dispatch:
+    inputs:
+      push:
+        description: Push gem
+        required: true
+        type: boolean
+        default: true
+  push:
+    branches:
+      - "**"
+
+env:
+  GEM_HOST: 'https://rubygems.pkg.github.com/DataDog'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build gem
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@31a7f6d628878b80bc63375a93ae079ec50a1601 # v1.143.0
+        with:
+          ruby-version: '3.2'
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - name: Patch version
+        run: |
+          # Obtain context information
+          git_ref='${{ github.ref }}'
+          git_branch="$(echo "${git_ref}" | sed -e 's#^refs/heads/##')"
+          git_sha='${{ github.sha }}'
+          gha_run_id='${{ github.run_id }}'
+
+          # Output info for CI debug
+          echo git_ref="${git_ref}"
+          echo git_branch="${git_branch}"
+          echo git_sha="${git_sha}"
```

Comment:
Minor: I noticed this results in really long versions (since the sha is quite big) -- gha4745388214.g2f501acf6b5a9d41bb800eb091474cf7961d6f62.use.github.packages.for.gems .

I would suggest shortening to the first [n = 12 chars](https://stackoverflow.com/questions/18134627/how-much-of-a-git-sha-is-generally-considered-necessary-to-uniquely-identify-a) :)

Reply:
I thought about that as well. I will do it.

I was also thinking about shortening the branch name by having it hashed + taking a prefix as well. This way we could keep the last part for humans but still be able to put a stable branch-originating segment anywhere we'd want (I was thinking about per branch grouping without having that big dotted name in the middle)

Reply:
I kinda like having the full branch name since it's more obvious a) it's not a regular release and b) what that release has in particular.